core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -125,15 +125,34 @@ private[spark] object SerDe {
   }
 
   def readDate(in: DataInputStream): Date = {
-    Date.valueOf(readString(in))
+    try {
+      val inStr = readString(in)
+      if (inStr == "NA") {
+        null
+      } else {
+        Date.valueOf(inStr)
+      }
+    } catch {
+      // On windows we get NegativeArraySizeException for NAs in R
```

Comment:
I dont think we should use this exception unless we can figure out why we `NegativeSizeExceptions` -- we haven't run into this for other types right ?